build/shimplify.py

```diff
@@ -0,0 +1,343 @@
+# Copyright (c) 2023, NVIDIA CORPORATION.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""implementation of shimplify Ant task in Python2.7 for Jython"""
+
+import errno
+import json
+import logging
+import os
+import re
+import subprocess
+
+# wrap Ant interactions to reduce linting warnings
+# TODO pass only Python types if possible
+
+
+def __project():
+    return project
+
+
+def __attributes():
+    return attributes
+
+
+# get Ant project property as string
+def __ant_proj_prop(name):
+    return str(__project().getProperty(name))
+
+
+def __ant_attr(name):
+    return str(__attributes().get(name))
+
+
+def __is_enabled_property(prop_name):
+    assert prop_name is not None, "Invalid property: None"
+    prop_val = __ant_proj_prop(prop_name)
+    return str(True).lower() == prop_val
+
+
+def __is_enabled_attr(attr):
+    assert attr is not None, "Invalid attribute: None"
+    attr_val = __ant_attr(attr)
+    return attr_val is not None and __is_enabled_property(attr_val)
+
+
+def __csv_ant_prop_as_arr(name):
+    prop_val = __ant_proj_prop(name)
+    return __csv_as_arr(prop_val)
+
+
+def __csv_attr_as_arr(name):
+    attr_val = __ant_attr(name)
+    return __csv_as_arr(attr_val)
+
+
+def __csv_as_arr(str_val):
+    if str_val in (None, ''):
+        return []
+    else:
+        return str_val.translate(None, ' \n\r').split(',')
+
+
+should_add_comment = __is_enabled_attr('if')
+
+# should we move files?
+should_move_files = __is_enabled_property('shimplifyMove')
+
+# allowed to overwrite the existing comment
+should_overwrite = __is_enabled_property('overShimplify')
```

Comment:
I think these option names could be a bit more verbose.  Took me longer than it should have to understand what `overShimplify` meant, had no idea until I saw this comment.  Also may be a bit better to have all the options under the `shimplify` top-level domain, i.e.: `shimplifyMove` --> `shimplify.move`, `overShimplify` --> `shimplify.overwrite`, etc. but that's just a preference thing, will defer to what you think is best.